src/nvim/diff.c

```diff
@@ -1870,6 +1870,35 @@ bool diffopt_horizontal(void)
   return (diff_flags & DIFF_HORIZONTAL) != 0;
 }
 
+// Compare the characters at "p1" and "p2".  If they are equal (possibly
+// ignoring case) return true and set "len" to the number of bytes.
+static bool diff_equal_char(const char_u *const p1, const char_u *const p2,
+                            int *const len)
+{
+  const int l = utfc_ptr2len(p1);
+
+  if (l != utfc_ptr2len(p2)) {
+    return false;
+  }
+  if (l > 1) {
+    if (STRNCMP(p1, p2, l) != 0
+        && (!enc_utf8
```

Comment:
You can drop use of `enc_utf8` as there is an effort to deprecate it.

Reply:
I'll do it in a separate commit so it's easier to review.